packages/runtime/container-runtime/src/summarizerClientElection.ts

```diff
@@ -6,88 +6,120 @@
+                    this.logger.sendErrorEvent({
+                        eventName: "UnexpectedElectionSequenceNumber",
+                        sequenceNumber,
+                        // Expected to be undefined
+                        lastSummaryAckSeqForClient: this.lastSummaryAckSeqForClient,
+                        // Expected to be same as op sequenceNumber
+                        electionSequenceNumber,
+                    });
+                }
             }
         });
 
+        // When a summary ack comes in, reset our op seq counter.
         this.summaryCollection.on(MessageType.SummaryAck, (op) => {
-            this.hasLoggedTelemetry = false;
             this.lastSummaryAckSeqForClient = op.sequenceNumber;
         });
 
-        this.clientElection.on("summarizerChange", (summarizerCount) => {
-            // Check if the number of summarizers in the quorum switches between zero
-            // and non-zero. That can indicate a change in whether we should summarize
-            // or not since we wait until the quorum no longer has any summarizers in
-            // the quorum before starting our own summarizer.
-            // Note this will be removed in a follow-up PR.
-            const previouslyHadAnySummarizersInQuorum = this._hasAnySummarizersInQuorum;
-            this._hasAnySummarizersInQuorum = summarizerCount > 0;
-            if (previouslyHadAnySummarizersInQuorum !== this._hasAnySummarizersInQuorum) {
-                this.emit("shouldSummarizeStateChanged");
-            }
-        });
-        this.clientElection.on("electedChange", (client: ITrackedClient | undefined) => {
-            this.hasLoggedTelemetry = false;
-            if (client !== undefined) {
-                // set to join seq
-                this.lastSummaryAckSeqForClient = client.sequenceNumber;
+        // Use oldest client election for unanimously and deterministically deciding
+        // which client should summarize.
+        this.clientElection.on("election", (client, sequenceNumber) => {
+            this.lastSummaryAckSeqForClient = undefined;
+            if (client === undefined && this.clientElection.eligibleCount > 0) {
+                // If no client is valid for election, reset to the oldest again.
+                // Also make extra sure not to get stuck in an infinite loop here:
+                // If there are no eligible clients, just wait until a client joins
+                // and will be auto-elected.
+                this.clientElection.resetElectedClient(sequenceNumber);
+                return;
             }
-            this.emit("shouldSummarizeStateChanged");
+            // Election can trigger a change in SummaryManager state.
+            this.emit("electedSummarizerChanged");
         });
-        this._hasAnySummarizersInQuorum = this.clientElection.getSummarizerCount() > 0;
+    }
+
+    public serialize(): ISerializedElection {
```

Comment:
the whole serialize/deserialize flow seems over complicated with deserialization decoupled in orderedClientElection and loadOrderedClients. can we just add a static create funciton to this class that takes the serialize form, and whatever else is needed, rather than spreading across muliple classes and funcitons

Reply:
So I can simplify it, which will probably make it easier to understand, but the intention in creating & loading both the OrderedClientCollection and all OrderedClientElections in a separate function is that it ensures they are instantiated together. It is important that they are created synchronously together, and this gives that direction via code.

Reply:
I changed it to just call all the constructors plainly in ContainerRuntime to show the difference, but I think it makes more sense to have the separate load function. The comparison is easy to see by looking at only this commit: https://github.com/microsoft/FluidFramework/pull/6335/commits/2a1dfb27952bc5e43bd10e9d2fbef7bbf7b6a45f